isis/src/kaguya/objs/KaguyaTcCamera/unitTest.cpp

```diff
@@ -0,0 +1,201 @@
+
+    qDebug() << "For lower left corner ...";
+    TestLineSamp(cam, 1.0, cam->Lines());
+
+    qDebug() << "For lower right corner ...";
+    TestLineSamp(cam, cam->Samples(), cam->Lines());
+
+    double samp = cam->Samples() / 2;
+    double line = cam->Lines() / 2;
+    qDebug() << "For center pixel position ...";
+
+    if(!cam->SetImage(samp, line)) {
+      qDebug() << "ERROR";
+      return 0;
+    }
+
+    if(abs(cam->UniversalLatitude() - knownLat) < 1E-13) {
+      qDebug() << "Latitude OK";
+    }
+    else {
+      qDebug() << qSetRealNumberPrecision(18)
+               << "Latitude off by: " << cam->UniversalLatitude() - knownLat;
+    }
+
+    if(abs(cam->UniversalLongitude() - knownLon) < 1E-11) {
+      qDebug() << "Longitude OK";
+    }
+    else {
+      qDebug() << qSetRealNumberPrecision(18)
+               << "Longitude off by: " << cam->UniversalLongitude() - knownLon;
+    }
+
+
+    qDebug() << "";
+    qDebug() << "";
+    qDebug() << "Testing TC1 s L2B0 image...";
+    knownLat = -82.0214347373044603;
+    knownLon = 46.4180177580203335;
+
+    Cube c2("$kaguya/testData/TC1S2B0_01_06691S820E0465.cub", "r");
+    KaguyaTcCamera *cam2 = (KaguyaTcCamera *) CameraFactory::Create(c2);
+    qDebug() << "FileName: " << FileName(c2.fileName()).name();
+    qDebug() << "CK Frame: " << cam2->instrumentRotation()->Frame();
+    qDebug() << "";
+
+    // Test kernel IDs
+    qDebug() << "Kernel IDs: ";
+    qDebug() << "CK Frame ID = " << cam2->CkFrameId();
+    qDebug() << "CK Reference ID = " << cam2->CkReferenceId();
+    qDebug() << "SPK Target ID = " << cam2->SpkTargetId();
+    qDebug() << "SPK Reference ID = " << cam2->SpkReferenceId();
+    qDebug() << "";
+
+    qDebug() << qSetRealNumberPrecision(18) << "Focal Length = " << cam2->FocalLength();
+    qDebug() << "";
+
+    // Test all four corners to make sure the conversions are right
+    qDebug() << "For upper left corner ...";
+    TestLineSamp(cam2, 1.0, 1.0);
+
+    qDebug() << "For upper right corner ...";
+    TestLineSamp(cam2, cam2->Samples(), 1.0);
+
+    qDebug() << "For lower left corner ...";
+    TestLineSamp(cam2, 1.0, cam2->Lines());
+
+    qDebug() << "For lower right corner ...";
+    TestLineSamp(cam2, cam2->Samples(), cam2->Lines());
+
+    samp = cam2->Samples() / 2;
+    line = cam2->Lines() / 2;
+    qDebug() << "For center pixel position ...";
+
+    if(!cam2->SetImage(samp, line)) {
+      qDebug() << "ERROR";
+      return 0;
+    }
+
+    if(abs(cam2->UniversalLatitude() - knownLat) < 1E-13) {
+      qDebug() << "Latitude OK";
+    }
+    else {
+      qDebug() << qSetRealNumberPrecision(18)
+               << "Latitude off by: " << cam2->UniversalLatitude() - knownLat;
+    }
+
+    if(abs(cam2->UniversalLongitude() - knownLon) < 1E-11) {
+      qDebug() << "Longitude OK";
+    }
+    else {
+      qDebug() << qSetRealNumberPrecision(18)
+               << "Longitude off by: " << cam2->UniversalLongitude() - knownLon;
```

Comment:
I really like all of your qDebug statements: they are going to make failed tests easier to parse later.

Reply:
:+1: